fundingmanager_test.go

```diff
@@ -0,0 +1,1046 @@
+
+	for _, msg := range announcements {
+		switch msg.(type) {
+		case *lnwire.ChannelAnnouncement:
+			gotChannelAnnouncement = true
+		case *lnwire.ChannelUpdate:
+			gotChannelUpdate = true
+		case *lnwire.AnnounceSignatures:
+			gotAnnounceSignatures = true
+		}
+	}
+
+	if !gotChannelAnnouncement {
+		t.Fatalf("did not get ChannelAnnouncement from Alice")
+	}
+	if !gotChannelUpdate {
+		t.Fatalf("did not get ChannelUpdate from Alice")
+	}
+	if !gotAnnounceSignatures {
+		t.Fatalf("did not get AnnounceSignatures from Alice")
+	}
+
+	// Do the check for Bob as well
+	announcements[0] = <-bobAnnounceChan
+	announcements[1] = <-bobAnnounceChan
+	announcements[2] = <-bobAnnounceChan
+
+	gotChannelAnnouncement = false
+	gotChannelUpdate = false
+	gotAnnounceSignatures = false
+
+	for _, msg := range announcements {
+		switch msg.(type) {
+		case *lnwire.ChannelAnnouncement:
+			gotChannelAnnouncement = true
+		case *lnwire.ChannelUpdate:
+			gotChannelUpdate = true
+		case *lnwire.AnnounceSignatures:
+			gotAnnounceSignatures = true
+		}
+	}
+
+	if !gotChannelAnnouncement {
+		t.Fatalf("did not get ChannelAnnouncement from Bob")
+	}
+	if !gotChannelUpdate {
+		t.Fatalf("did not get ChannelUpdate from Bob")
+	}
+	if !gotAnnounceSignatures {
+		t.Fatalf("did not get AnnounceSignatures from Bob")
+	}
+
+	// The funding process is now finished, wait for the
+	// OpenStatusUpdate_ChanOpen update
+	openUpdate := <-updateChan
+	_, ok := openUpdate.Update.(*lnrpc.OpenStatusUpdate_ChanOpen)
+	if !ok {
+		t.Fatal("OpenStatusUpdate was not OpenStatusUpdate_ChanOpen")
+	}
+
+	// The internal state-machine should now have deleted the channelStates from
+	// the database, as the channel is announced.
+	state, _, err = aliceFundingMgr.getChannelOpeningState(fundingOutPoint)
+	if err != channeldb.ErrChannelNotFound {
+		t.Fatalf("expected to not find channel state, but got: %v", state)
+	}
+
+	// Need to give bob time to update database.
+	time.Sleep(100 * time.Millisecond)
+
+	state, _, err = bobFundingMgr.getChannelOpeningState(fundingOutPoint)
+	if err != channeldb.ErrChannelNotFound {
+		t.Fatalf("expected to not find channel state, but got: %v", state)
+	}
+
+}
+
+func TestFundingManagerRestartBahavior(t *testing.T) {
+	setupFundingManagers(t)
+	defer tearDownFundingManagers()
+
+	// Run through the process of opening the channel, up until the funding
+	// transaction is broadcasted.
+	openChannel(t)
+
+	// After the funding transaction gets mined, both nodes will send the
+	// fundingLocked message to the other peer. If the funding node fails before
+	// this message has been successfully sent, it should retry sending it on
+	// restart. We mimic this behavior by letting the SendToPeer method return an
+	// error, as if the message was not successfully sent. we then recreate the
+	// fundingManager and make sure it continus the process as expected.
+	aliceFundingMgr.cfg.SendToPeer = func(target *btcec.PublicKey, msgs ...lnwire.Message) error {
```

Comment:
Discussed offline, and will add such a callback mechanism to the `server`, then do the follow up changes to `fundingManager` in a later PR. Added a `todo` in `fundingManager.Start()`